pkg/groupsync/errors.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupsync
+
+import "errors"
+
+// ErrTargetUserIDNotFound denotes when the user ID for the target system cannot be found.
+var ErrTargetUserIDNotFound = errors.New("target user ID not found")
```

Comment:
```suggestion
type Error string

func (e Error) Error() string {
  return string(e)
}

// ErrTargetUserIDNotFound denotes when the user ID for the target system cannot be found.
const ErrTargetUserIDNotFound = Error("target user ID not found")
```

Reply:
what are the advantages of doing it this way?

Reply:
Global variables can be modified by any imported package. It's a common security issue. Constants cannot (and are slightly more efficient, but that's not the real optimization).